### PR TITLE
Add integration tests for StreamLayerClient Subscribe

### PR DESCRIPTION
**Pull request by dpoberezh**
Resolves: OLPEDGE-1469

Signed-off-by: Dmytro Poberezhnyi <ext-dmytro.poberezhnyi@here.com>

### CODE REVIEW

File: tests/integration/olp-cpp-sdk-dataservice-read/StreamLayerClientTest.cpp [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+                << ", status: " << error.GetHttpStatusCode()
+                << ", message: " << error.GetMessage();
+  return result_stream.str();
+}
+
+void ReadStreamLayerClientTest::SetUp() {
+  network_mock_ = std::make_shared<NetworkMock>();
+  settings_ = OlpClientSettings();
+  settings_.network_request_handler = network_mock_;
+  olp::cache::CacheSettings cache = {};
```
Comment (borzun) [inline tests/integration/olp-cpp-sdk-dataservice-read/StreamLayerClientTest.cpp]: Do you use this variable?

File: tests/integration/olp-cpp-sdk-dataservice-read/StreamLayerClientTest.cpp [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gmock/gmock.h>
+#include <matchers/NetworkUrlMatchers.h>
+#include <mocks/NetworkMock.h>
+#include <olp/core/cache/DefaultCache.h>
```
Comment (borzun) [inline tests/integration/olp-cpp-sdk-dataservice-read/StreamLayerClientTest.cpp]:
```suggestion
```
Looks like you don't need this include.

File: tests/integration/olp-cpp-sdk-dataservice-read/StreamLayerClientTest.cpp [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+      condition.Notify();
+    });
+
+    ASSERT_TRUE(condition.Wait(kTimeout));
+
+    ASSERT_TRUE(subscribe_response.IsSuccessful())
+        << ApiErrorToString(subscribe_response.GetError());
+
+    EXPECT_EQ(kSubscriptionId, subscribe_response.GetResult().c_str());
+  }
```
Comment (borzun) [inline tests/integration/olp-cpp-sdk-dataservice-read/StreamLayerClientTest.cpp]: I guess you should clear and verify the expectations of network after each `SCOPED_TRACE`

**Comment by meravingen123** [comment]
Please add test for the invalid hrn and the invalid layer cases.

File: tests/integration/olp-cpp-sdk-dataservice-read/StreamLayerClientTest.cpp (hunk 3 above)
Comment (dpoberezh) [reply tests/integration/olp-cpp-sdk-dataservice-read/StreamLayerClientTest.cpp]: Added

**Comment by dpoberezh** [comment]
> Please add test for the invalid hrn and the invalid layer cases.

Added, please see
"Subscribe fails, incorrect hrn"
"Subscribe fails, incorrect layer"

**Pull request merged**